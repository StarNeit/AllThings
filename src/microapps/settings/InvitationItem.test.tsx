import { addYears, subYears } from 'date-fns/fp'
import React from 'react'
import {
  InvitationItemCreatedAtMessage,
  InvitationItemExpiresAtMessage,
} from './InvitationItem'

// Needed as localizeDate returns a different time on the CI!
jest.mock('../../components/DateString', () => ({
  localizeDate: jest.fn(() => 'February 15, 2017, 1:00 AM'),
}))

describe('Check invitation items', () => {
  it('should display one for created at', () => {
    const wrapper = global.mountIntl(
      <InvitationItemCreatedAtMessage createdAt={new Date().toISOString()} />,
    )
    expect(wrapper).toMatchSnapshot()
  })

  it('should display one for expires at - no expiration date', () => {
    const wrapper = global.mountIntl(<InvitationItemExpiresAtMessage />)
    expect(wrapper).toMatchSnapshot()
  })

  it('should display one for expires at - past expiration date', () => {
    const invitation = {
      expiresAt: subYears(10)(new Date()).toISOString(),
    }
    const wrapper = global.mountIntl(
      <InvitationItemExpiresAtMessage {...invitation} />,
    )
    expect(wrapper).toMatchSnapshot()
  })

  it('should display one for expires at - future expiration date', () => {
    const invitation = {
      expiresAt: addYears(10)(new Date()).toISOString(),
    }
    const wrapper = global.mountIntl(
      <InvitationItemExpiresAtMessage {...invitation} />,
    )
    expect(wrapper).toMatchSnapshot()
  })
})
