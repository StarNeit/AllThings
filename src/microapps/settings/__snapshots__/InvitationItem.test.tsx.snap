// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Check invitation items should display one for created at 1`] = `
<span>
  Invited: February 15, 2017, 1:00 AM
</span>
`;

exports[`Check invitation items should display one for expires at - future expiration date 1`] = `
.css-0,
[data-css-0] {
  display: block;
  font-family: "Open Sans", Helvetica, Arial, sans-serif;
  font-weight: 400;
  font-size: 14px;
  -webkit-font-smoothing: antialiased;
  -moz-osx-font-smoothing: grayscale;
  color: #333333;
}

.css-1,
[data-css-1] {
  box-sizing: border-box;
  flex: 0 0 auto;
  -webkit-flex: 0 0 auto;
}

<ForwardRef
  data-css-0=""
  style={
    Object {
      "color": "#626262",
    }
  }
>
  <div
    data-css-1=""
    data-css-0=""
    style={
      Object {
        "color": "#626262",
      }
    }
  >
    <span>
      Expires in 10 years
    </span>
  </div>
</ForwardRef>
`;

exports[`Check invitation items should display one for expires at - no expiration date 1`] = `
.css-0,
[data-css-0] {
  display: block;
  font-family: "Open Sans", Helvetica, Arial, sans-serif;
  font-weight: 400;
  font-size: 14px;
  -webkit-font-smoothing: antialiased;
  -moz-osx-font-smoothing: grayscale;
  color: #333333;
}

.css-1,
[data-css-1] {
  box-sizing: border-box;
  flex: 0 0 auto;
  -webkit-flex: 0 0 auto;
}

<ForwardRef
  data-css-0=""
  style={
    Object {
      "color": "#626262",
    }
  }
>
  <div
    data-css-1=""
    data-css-0=""
    style={
      Object {
        "color": "#626262",
      }
    }
  >
    <span>
      No expiration date
    </span>
  </div>
</ForwardRef>
`;

exports[`Check invitation items should display one for expires at - past expiration date 1`] = `
.css-0,
[data-css-0] {
  display: block;
  font-family: "Open Sans", Helvetica, Arial, sans-serif;
  font-weight: 400;
  font-size: 14px;
  -webkit-font-smoothing: antialiased;
  -moz-osx-font-smoothing: grayscale;
  color: #333333;
}

.css-1,
[data-css-1] {
  box-sizing: border-box;
  flex: 0 0 auto;
  -webkit-flex: 0 0 auto;
}

<ForwardRef
  data-css-0=""
  style={
    Object {
      "color": "#626262",
    }
  }
>
  <div
    data-css-1=""
    data-css-0=""
    style={
      Object {
        "color": "#626262",
      }
    }
  >
    <span>
      Expired 10 years ago
    </span>
  </div>
</ForwardRef>
`;
